apps/hash-frontend/src/pages/flows.page/flow-definition/topbar.tsx

```diff
@@ -0,0 +1,161 @@
+import {
+  AngleRightRegularIcon,
+  InfinityLightIcon,
+  PlayIconSolid,
+  Select,
+} from "@hashintel/design-system";
+import type { SxProps, Theme } from "@mui/material";
+import { Box, outlinedInputClasses, Stack, Typography } from "@mui/material";
+import { format } from "date-fns";
+import { useMemo } from "react";
+
+import { Button } from "../../../shared/ui/button";
+import { MenuItem } from "../../../shared/ui/menu-item";
+import { useFlowDefinitionsContext } from "./shared/flow-definitions-context";
+import { useFlowRunsContext } from "./shared/flow-runs-context";
+
+const typographySx: SxProps<Theme> = {
+  color: ({ palette }) => palette.gray[70],
+  fontWeight: 500,
+  lineHeight: 1,
+};
+
+const selectSx: SxProps<Theme> = {
+  background: "transparent",
+  boxShadow: "none",
+  [`& .${outlinedInputClasses.input}`]: {
+    fontSize: 14,
+    padding: "6px 12px",
+    ...typographySx,
+  },
+  "& svg": {
+    fontSize: 14,
+  },
+};
+
+const Divider = () => (
+  <AngleRightRegularIcon
+    sx={{
+      fill: ({ palette }) => palette.gray[50],
+      fontSize: 18,
+      mr: 0.5,
+      ml: 1,
+    }}
+  />
+);
+
+export const Topbar = ({
+  handleRunFlowClicked,
+}: {
+  handleRunFlowClicked: () => void;
+}) => {
+  const { flowDefinitions, selectedFlow, setSelectedFlow } =
+    useFlowDefinitionsContext();
+
+  const { flowRuns, selectedFlowRun, setSelectedFlowRun } =
+    useFlowRunsContext();
+
+  const runOptions = useMemo(
+    () =>
+      flowRuns.filter(
+        (run) => run.inputs[0].flowDefinition.name === selectedFlow.name,
+      ),
+    [flowRuns, selectedFlow.name],
+  );
```

Comment:
Possibly this is similar to the comment I left in a prior PR, but just want to make you aware when I clicked `Run` on a flow where there are no prior runs the list of runs didn't appear for me for a few seconds. Maybe this is something we can address once we actually switch to the active run after the button is pressed, when we change what the `startFlow` resolver returns.

Reply:
Yes – I can do this as a drive-by in any upcoming task if you don't need the `startFlow` resolver to return the outputs any more